### PR TITLE
[handler] log the real client address

### PR DESCRIPTION
**Pull request by qingwave**
Log the real client address more than proxy ip.

### CODE REVIEW

File: src/app/backend/handler/filter.go [hunk 1]
```diff
@@ -177,3 +177,33 @@ func mapUrlToResource(url string) *string {
 	}
 	return &parts[3]
 }
+
+// remoteAddr extracts the remote address of the request, taking into
+// account proxy headers.
+func remoteAddr(r *http.Request) string {
```
Comment (maciaszczykm) [inline src/app/backend/handler/filter.go:183]: Please change to `getRemoteAddr`.

File: src/app/backend/handler/filter.go [hunk 2]
```diff
@@ -177,3 +177,33 @@ func mapUrlToResource(url string) *string {
...
+		if len(proxies) > 0 {
+			remoteAddr := strings.TrimSpace(proxies[0])
+			if remoteAddr != "" {
+				return remoteAddr
+			}
+		}
+	}
+
+	if prior := r.Header.Get("X-Forwarded-For"); prior != "" {
+		proxies := strings.Split(prior, ",")
```
Comment (maciaszczykm) [inline src/app/backend/handler/filter.go:195]: The part inside `if` statement is duplicated, please export it to the new function.

File: src/app/backend/handler/filter.go [hunk 3]
```diff
@@ -177,3 +177,33 @@ func mapUrlToResource(url string) *string {
 	}
 	return &parts[3]
 }
+
+// remoteAddr extracts the remote address of the request, taking into
+// account proxy headers.
+func remoteAddr(r *http.Request) string {
+	if prior := r.Header.Get("X-Original-Forwarded-For"); prior != "" {
```
Comment (floreks) [inline src/app/backend/handler/filter.go:184]: Export those header strings to a custom named string type

Comment (qingwave) [reply src/app/backend/handler/filter.go]: fixed

File: src/app/backend/handler/filter.go (hunk 2 above)
Comment (qingwave) [reply src/app/backend/handler/filter.go]: fixed

**Pull request merged**